lib/matplotlib/tests/test_axes.py

```diff
@@ -4863,6 +4863,25 @@ def test_pandas_indexing_hist():
     axes.hist(ser_2)
 
 
+def test_pandas_bar_align_center():
+    # Tests fix for issue 8767
+    pd = pytest.importorskip('pandas')
+
+    df = pd.DataFrame({'col1': [1] + [2],
+                       'col2': [1]*2,
+                       'col3': [1]*2})
+
+    fig, axArr = plt.subplots(2)
+
+    curr = 1
+    for ax in axArr:
```

Comment:
The trick is probably to have an index without 0 in it, I think `df = pd.DataFrame({'a': range(5), 'b': range(5)}, index=list('abcde'))` would produce the exception.